### ISSUE TITLE
How to properly build GUI on Windows?

### ISSUE THREAD

**Issue by DanH-ER** [MediaArea/MediaInfo#788]
When I build the Visual Studio project on Windows the resulting GUI app has the sheets and tree views disabled, is missing the preferences window, and just generally looks different that the official release from the website. So my question is how do I build the same UI that's in the official release? Is there a config or build flag step I'm missing?

The instructions in the readme only describe how to build Mac and Linux, not Windows.

**Comment by DanH-ER** [comment]
I downloaded the community edition of Embarcadero and I can build the UI in 32 bit mode but when I try to build the 64 bit version I get errors similar to this...

[ilink64 Error] Error: Unresolved external 'ZenLib::ZtringListListF::ZtringListListF()' referenced from C:\REPOS\MEDIAINFO\PROJECT\BCB\GUI\WIN64\RELEASE\PREFERENCES.O

I built the BCB ZenLib project first, in both 32 and 64 bit, so I'm not sure what is causing this. 

Also I can't seem to build MediaInfoLib at all. The 32 bit version throws errors complaining about int128u and the the 64 bit version complains about __uuifof. 

Since you guys use this for the official build I assume you don't have these errors. Any tips on what might be causing them?

**Comment by DanH-ER** [comment]
Never mind I see that the release script uses the 32bit GUI and the DLLs from Visual Studio. I think I see how it works.

**Comment by cjee21** [comment]
Documenting how I managed to build the Windows GUI here to help others and for my future reference. Thanks to https://github.com/MediaArea/MediaInfo/issues/788#issuecomment-1887569305 and https://github.com/MediaArea/MediaInfo/issues/788#issuecomment-1888203511 for giving me an idea.

### How to build Windows GUI
1. Install the following required software on a Windows PC:
   - C++Builder Community Edition (https://www.embarcadero.com/products/cbuilder/starter)
   - Git for Windows (https://gitforwindows.org/)
2. Create an empty folder. We'll call this `MediaInfoDevelopment` for this guide.
3. Clone the following MediaInfo dependencies to `MediaInfoDevelopment` using Git:
   - https://github.com/MediaArea/MediaInfoLib
   - https://github.com/MediaArea/ZenLib
   - https://github.com/MediaArea/zlib
4. Clone MediaInfo (this repository) to `MediaInfoDevelopment`.
   - If you intend to contribute to MediaInfo, fork the repository on GitHub first and clone your fork instead.
5. The `MediaInfoDevelopment` folder should now have the following folders:
   - `MediaInfo`
   - `MediaInfoLib`
   - `ZenLib`
   - `zlib`
6. Build MediaInfo dependencies by opening the following in C++Builder and selecting Project>Build:
   - `MediaInfoDevelopment\ZenLib\Project\BCB\ZenLib_Proj.groupproj`
   - `MediaInfoDevelopment\zlib\contrib\BCB\zlib_Proj.groupproj`
 7. Copy `MediaInfo_i386.dll` from the [official distribution](https://mediaarea.net/en/MediaInfo/Download/Windows) to `MediaInfoDevelopment\MediaInfo\Project\BCB\GUI\Win32\Release` unless you want to build it yourself.
 8. Open `MediaInfoDevelopment\MediaInfo\Project\BCB\MediaInfo_Proj.groupproj` in C++Builder.
 9. Build MediaInfo GUI by selecting Project>Build.
 10. You should now be able to Run MediaInfo and see the GUI appear.

**Comment by cjee21** [comment]
> I downloaded the community edition of Embarcadero and I can build the UI in 32 bit mode but when I try to build the 64 bit version I get errors similar to this...
> 
> [ilink64 Error] Error: Unresolved external 'ZenLib::ZtringListListF::ZtringListListF()' referenced from C:\REPOS\MEDIAINFO\PROJECT\BCB\GUI\WIN64\RELEASE\PREFERENCES.O
> 
> I built the BCB ZenLib project first, in both 32 and 64 bit, so I'm not sure what is causing this.

I successfully built a 64-bit version. Anyone interested? 😄 

![Screenshot 2024-06-22 180755](https://github.com/MediaArea/MediaInfo/assets/77721854/3e38e973-e21d-46b8-9ad7-e9e6a591c65a)
![Screenshot 2024-06-22 180653](https://github.com/MediaArea/MediaInfo/assets/77721854/27b963c0-b200-4f9c-9a53-9b93dc50ff93)

**Comment by JeromeMartinez** [comment]
> I successfully built a 64-bit version. Anyone interested? 😄

Did you need to change the project files?
If so, please do a PR.
And we could put the 64-bit version in the 64-bit 7z package without installer, unfortunately 32-bit is still there and for the more used installer it would inflate the size to have both 32 and 64 bit, so not now, but at long term we may put the 64-bit version in the installer and move the 32-bit installer to a 2nd choice when the installer detects 64-bit...

**Comment by cjee21** [comment]
> Did you need to change the project files? If so, please do a PR. 

Yes, changed project files for MediaInfo BCB, ZenLib BCB and zlib BCB.

> And we could put the 64-bit version in the 64-bit 7z package without installer, 

This should be easy to include in the PR.

>unfortunately 32-bit is still there and for the more used installer it would inflate the size to have both 32 and 64 bit, so not now, but at long term we may put the 64-bit version in the installer and move the 32-bit installer to a 2nd choice when the installer detects 64-bit...

I can try and see how much it inflates...

**Comment by JeromeMartinez** [comment]
> I can try and see how much it inflates...

Package size is a hot topic for us, most people don't care at all and download GB of content but some others complain loudly when there is 1 more MB in the main package...

For the 7z, it would actually helps as we can remove the 32-bit DLL in that case.

**Comment by cjee21** [comment]
> > I can try and see how much it inflates...
> 
> Package size is a hot topic for us, most people don't care at all and download GB of content but some others complain loudly when there is 1 more MB in the main package...

Then maybe better to make two installers. Most PCs should be 64-bit now and Windows 11 does not even have a 32-bit version. Now it also contains both 64-bit and 32-bit DLLs.

Good thing the WebView2Loader.dll is only 100+KB

**Comment by cjee21** [comment]
Making an installer that handles both 32-bit and 64-bit is around 9MB

**Comment by cjee21** [comment]
FYI updated the guide above (https://github.com/MediaArea/MediaInfo/issues/788#issuecomment-2124537221). Will update it again when the PRs are merged.

**Comment by JeromeMartinez** [comment]
> FYI updated the guide above

Thank you!

**Comment by cjee21** [comment]
FYI guide (https://github.com/MediaArea/MediaInfo/issues/788#issuecomment-2124537221) updated again.

**Comment by cjee21** [comment]
First bug discovered. If open a file from explorer context menu then immediately switch from Basic to Graph view, this pops up:
![Screenshot 2024-07-03 120730](https://github.com/MediaArea/MediaInfo/assets/77721854/6286e478-c621-4ee0-b9ff-57cfae5c9226)

If switch to HTML view then to Graph view, it works.

**Comment by cjee21** [comment]
Updated the guide (https://github.com/MediaArea/MediaInfo/issues/788#issuecomment-2124537221) as there are many changes to MediaInfo recently.
